csharp/tools/tracing-config.lua

```diff
@@ -31,19 +32,30 @@ function RegisterExtractorPack(id)
             local firstCharacter = string.sub(arg, 1, 1)
             if not (firstCharacter == '-') and not (firstCharacter == '/') then
                 Log(1, 'Dotnet subcommand detected: %s', arg)
-                if arg == 'build' or arg == 'msbuild' or arg == 'publish' or arg == 'pack' or arg == 'test' or
-                    arg == 'run' then match = true end
+                if arg == 'build' or arg == 'msbuild' or arg == 'publish' or arg == 'pack' or arg == 'test' then
+                    match = true
+                    break
+                end
+                if arg == 'run' then
+                    -- for `dotnet run`, we need to make sure that `/p:UseSharedCompilation=false` is
+                    -- not passed in as an argument to the program that is run
+                    match = true
+                    needsSeparator = true
+                end
+            end
+            if arg == '--' then
+                needsSeparator = false
                 break
             end
         end
         if match then
+            local injections = { '/p:UseSharedCompilation=false' }
+            if needsSeparator then
+                table.insert(injections, '--')
```

Comment:
For `dotnet run` you append `--` after the UseSharedCompilation flag (if not otherwise provided in the list of arguments). Does `--` have special semantic meaning, when executing `dotnet run`. That is, are everything *after* `--` considered arguments to the program that is run?

Reply:
Yes, indeed:
```
Description:
  .NET Run Command

Usage:
  dotnet run [options] [[--] <additional arguments>...]]

Additional Arguments:
  Arguments passed to the application that is being run.
```